### DISCUSSION TITLE
Camera zoom

### DISCUSSION THREAD

**Discussion by gmoller** [bevyengine/bevy#2849] [Q&A] [answered]
Hi, I'm trying to use the Camera to be able to zoom in and out using the mouse wheel. I'm using the following code that I found online, but the update translation line does not compile:

pub(crate) fn move_camera_system(
    mut mouse_wheel_events: EventReader<MouseWheel>,
    mut query: Query<&mut Transform, With<bevy::render::camera::Camera>>
) {

    for mut transform in query.iter_mut() {
        // Camera Mouse Zoom
        for event in mouse_wheel_events.iter() {
            // this does not compile (transform can't be multiplied by a Vec3):
            transform.translation -= transform * Vec3::ONE * event.y * 1.0; // zoom_sensitivity = 1.0
        }
    }
}

Any help would be greatly appreciated.

Thanks,
Greg

**Comment by BlackPhlox** [comment, marked as answer]
The compiler actually tells you whats wrong `transform.translation -= transform.translation *...` cause transfrom contains translation, rotation and scale
